src/lib/lightning/clightning/clightningService.ts

```diff
@@ -197,6 +199,43 @@ class CLightningService implements LightningService {
     );
   }
 
+  async addListenerToNode(node: LightningNode): Promise<void> {
+    console.log(`addListenerToNode ${node.implementation}`);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async removeListener(node: LightningNode): Promise<void> {
+    if (this.channelsInterval !== null) {
+      clearInterval(this.channelsInterval);
+    }
+  }
+
+  async subscribeChannelEvents(
+    node: LightningNode,
+    callback: (event: PLN.LightningNodeChannelEvent) => void,
+  ): Promise<void> {
+    // check c-lightning channels every 30 seconds
+    this.channelsInterval = setInterval(() => {
+      this.checkChannels(node, callback);
+    }, 3 * 10000);
+  }
+
+  checkChannels = async (
+    node: LightningNode,
+    callback: (event: PLN.LightningNodeChannelEvent) => void,
+  ) => {
+    const channels = await this.getChannels(node);
+    channels.forEach(channel => {
+      if (channel.pending) {
+        callback({ type: 'Pending' });
+      } else if (channel.status === 'Open') {
+        callback({ type: 'Open' });
+      } else if (channel.status === 'Closed') {
+        callback({ type: 'Closed' });
+      }
+    });
```

Comment:
This is going to call the callback every 30secs for all channels the node has. We are trying to mimic the behavior of subscribing to channel events which will only notify us when something changes with the node's channels. If the channels all have the same status as the previous check, it shouldn't call the callback because nothing has changed.